react-task/src/components/Cards/CardsCreationForm.js

```diff
@@ -0,0 +1,97 @@
+import PropTypes from 'prop-types';
+import { useState, useCallback, useRef} from 'react';
+import styles from './CardsCreationForm.module.scss';
+
+function CardsCreationForm({ cardAdd }) {
+  const title = useRef(null);
+  const price = useRef(null);
+  const imageUrl = useRef(null);
+  const description = useRef(null);
+
+  const [inputs, setInputs] = useState({
+    title: '',
+    price: 1,
+    description: '',
+    imageUrl: ''
+  });
+  const [inputWarning, setInputWarning] = useState(() => null);
+
+  const checkInputComplete = useCallback(() => {
+    let error = null;
+    const refs = [title, price, imageUrl, description];
+    for (const input in inputs) {
+      if (!inputs[input]) {
+        const element = refs.find(ref => ref.current.name === input).current;
+        error = {
+          sourceInput: element,
+          style: {
+            top: element.getBoundingClientRect().top,
+            left: element.getBoundingClientRect().left,
+            width: element.getBoundingClientRect().width,
+            height: element.getBoundingClientRect().height
+          }
+        };
+        break;
+      }
+    }
+    return error;
+  }, [inputs]);
+
+  const errorMessageClick = () => {
+    inputWarning.sourceInput.focus();
+    setInputWarning(null);
+  }
+
+  const btnAddClick = useCallback((event) => {
+    event.preventDefault();
+    const error = checkInputComplete();
+    setInputWarning(error);
+    if (!error) {
+      const card = {
+        title: inputs.title,
+        description: inputs.description,
+        price: Number.parseFloat(inputs.price),
+        imageUrl: inputs.imageUrl
+      }
+      cardAdd(card);
+    }
+  }, [inputs, cardAdd, checkInputComplete]);
+
+  const inputChange = (event) => {
+    const {name, value} = event.currentTarget;
+    setInputs({...inputs, [name]: value});
+  }
+
+  return (
+    <form className={styles.form} onSubmit={btnAddClick}>
+      <div className={styles.form__left}>
+        <div className={styles.form__input}>
```

Comment:
I think will be better if you create separate component  for example `InputField` and implement the main logic for it. And you can pass `title`, `onChange` and so on as a props.